hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieParquetInputFormat.java

```diff
@@ -91,9 +94,42 @@ private void initAvroInputFormat() {
     }
   }
 
+  private static boolean checkIfHudiTable(final InputSplit split, final JobConf job) {
+    try {
+      Option<Path> tablePathOpt = TablePathUtils.getTablePath(((FileSplit) split).getPath(), job);
+      if (!tablePathOpt.isPresent()) {
+        return false;
+      }
+      return tablePathOpt.get().getFileSystem(job).exists(new Path(tablePathOpt.get(), HoodieTableMetaClient.METAFOLDER_NAME));
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
   @Override
   public RecordReader<NullWritable, ArrayWritable> getRecordReader(final InputSplit split, final JobConf job,
                                                                    final Reporter reporter) throws IOException {
+
+    if (HoodieFileGroupReaderRecordReader.useFilegroupReader(job)) {
+      try {
+        if (!(split instanceof FileSplit) || !checkIfHudiTable(split, job)) {
+          return super.getRecordReader(split, job, reporter);
+        }
+        if (supportAvroRead && HoodieColumnProjectionUtils.supportTimestamp(job)) {
+          return new HoodieFileGroupReaderRecordReader((s, j, r) -> {
+            try {
+              return new ParquetRecordReaderWrapper(new HoodieTimestampAwareParquetInputFormat(), s, j, r);
+            } catch (InterruptedException e) {
+              throw new RuntimeException(e);
+            }
+          }, split, job, reporter);
+        } else {
+          return new HoodieFileGroupReaderRecordReader(super::getRecordReader, split, job, reporter);
```

Comment:
I test with hive3.1.2, partition query returns empty result with this reader.

Reply:
we can move https://github.com/apache/hudi/blob/2c38ef740d3d34e9eb05b59fa147c55623b81a90/hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieParquetInputFormat.java#L150-L156 to the top of this method.

Reply:
I added your suggestion. Could you please let me know if that fixes the issue? Thanks!

Reply:
Fixed.